thorlcr/activities/indexread/thindexreadslave.cpp

```diff
@@ -325,13 +327,17 @@ class CIndexReadSlaveBase : public CSlaveActivity
                     if (translator)
                         klManager->setLayoutTranslator(&translator->queryTranslator());
                     translators.append(translator.getClear());
-                    keyManagers.append(*klManager.getLink());
+                    {
+                        CriticalBlock b(keyManagersCS);
+                        keyManagers.append(*klManager.getLink());
+                    }
                     keyManager = klManager;
-                    partNum = p;
+                    partNum = (p+1);
```

Comment:
Change relates to: https://github.com/hpcc-systems/HPCC-Platform/pull/18144/commits/55ef2b94c7d452dbec6d168ce3e3f9cee8cca2c4#r1426670630